tensorflow/contrib/framework/python/ops/arg_scope.py

```diff
@@ -181,7 +181,7 @@ def func_with_args(*args, **kwargs):
     return func(*args, **current_args)
   _add_op(func)
   setattr(func_with_args, '_key_op', _key_op(func))
-  return tf_decorator.make_decorator(func, func_with_args)
+  return tf_decorator.make_decorator(func, func_with_args, decorator_name='add_arg_scope')
```

Comment:
Is this necessary? I.e. if this line is required, it means that `tf_decorator.make_decorator` isn't correctly getting the caller's name (`func_with_args`), which means that the new behavior isn't  working as intended.

Reply:
Passing in the decorator_name could be left out, and you're right - if this is the only call path that triggers the name look-up it could leave the changed behavior untested.  By my reading, the unit tests still cover inspection, and I will confirm other make_decorator calls as well.  

Originally my question was - why run additional call stack inspection when the caller could pass in its name? But after finding alternatives to `inspect.stack()` the overhead of the repeated lookup is no longer visible.

Long story short, I'll drop this change from my diff to ensure my main changes are well exercised.  Also - check into #13232 which is looking to provide this same fix to add_arg_scope.

Reply:
Should we add a test to document that behavior? (in another PR)

Reply:
Not a huge deal to me, I just want to confirm that the changes to line 184 aren't /required/.

Reply:
also @drpngx this functionality is tested, which is why I was curious about the change. Like I said, though, it's not important. I don't think it should hold up the PR.

Reply:
@JettJones the functionality is there to minimize the amount of duplication required to get decorators to work. It's lower friction than having to write `def my_decorator():` and then later `my_decorator`: string duplication is resistant to rename refactoring, etc. I'm not religious about it, it's just a convenience.

Reply:
I didn't find another caller of `make_decorator` under contrib, so I rolled back passing in `decorator_name` explicitly.  Thanks for calling this out.